### PR TITLE
feat(flink): add remaining operators for Flink to pass/skip the common tests

### PR DESCRIPTION
**Pull request by mfatihaktas**
* This is a follow-up to this [previously merged PR](https://github.com/ibis-project/ibis/pull/7265).
* Contains the remaining changes in `ibis/ibis/backends/flink/registry.py` to pass/skip the common tests for Flink backend.
* Next step is the [PR](https://github.com/ibis-project/ibis/pull/7366) that contains changes in the common tests to make them pass/skip for Flink backend.

### CODE REVIEW

File: ibis/backends/flink/registry.py [hunk 1]
```diff
@@ -40,9 +40,21 @@ def extract_field_formatter(translator: ExprTranslator, op: ops.Node) -> str:
...
+    if isinstance(to, Timestamp):
+        if arg.dtype.is_numeric():
+            arg_translated = f"FROM_UNIXTIME({arg_translated})"
+
+        if to.timezone:
+            return f"TO_TIMESTAMP(CONVERT_TZ(CAST({arg_translated} AS STRING), 'UTC+0', '{to.timezone}'))"
+        else:
+            return f"TO_TIMESTAMP({arg_translated})"
+
+    elif isinstance(to, JSON):
```
Comment (cpcloud) [inline ibis/backends/flink/registry.py:56]: Don't check with `isinstance`, use the is_* methods. A method exists for every leaf type as well as every parent class.

```suggestion
    from ibis.expr.datatypes.core import JSON, Timestamp

    arg, to = op.arg, op.to
    arg_translated = translator.translate(arg)
    if to.is_timestamp():
        if arg.dtype.is_numeric():
            arg_translated = f"FROM_UNIXTIME({arg_translated})"

        if to.timezone:
            return f"TO_TIMESTAMP(CONVERT_TZ(CAST({arg_translated} AS STRING), 'UTC+0', '{to.timezone}'))"
        else:
            return f"TO_TIMESTAMP({arg_translated})"

    elif to.is_json():
```

File: ibis/backends/flink/registry.py [hunk 2]
```diff
@@ -281,10 +251,37 @@ def _array_index(translator: ExprTranslator, op: ops.arrays.ArrayIndex):
...
+    key_array = translator.translate(op.keys)
+    value_array = translator.translate(op.values)
+
+    return f"MAP_FROM_ARRAYS({key_array}, {value_array})"
+
+
+def _map_get(translator: ExprTranslator, op: ops.maps.MapGet) -> str:
+    map_ = translator.translate(op.arg)
+    key = translator.translate(op.key)
+    return f"{map_} '[' {key} ']'"
```
Comment (cpcloud) [inline ibis/backends/flink/registry.py:280]: This looks like the same issue we found when compiling array indexing. Can you remove the single quotes surrounding the brackets?

File: ibis/backends/flink/registry.py [hunk 3]
```diff
@@ -281,10 +251,37 @@ def _array_index(translator: ExprTranslator, op: ops.arrays.ArrayIndex):
...
+
+
+def _day_of_week_index(translator: ExprTranslator, op: ops.Node) -> str:
+    arg = translator.translate(op.args[0])
```
Comment (cpcloud) [inline ibis/backends/flink/registry.py:284]: Remember to use the name of the argument and not `op.args[index]`.

File: ibis/backends/flink/registry.py [hunk 4]
```diff
@@ -319,6 +319,64 @@ def _string_to_timestamp(
     return f"TO_TIMESTAMP({arg}, {format_string})"
 
 
+def _time(translator: ExprTranslator, op: ops.temporal.Time) -> str:
+    from ibis.expr.datatypes.core import Timestamp
+
+    if isinstance(op.arg.dtype, Timestamp):
```
Comment (cpcloud) [inline ibis/backends/flink/registry.py:325]: Use the `is_timestamp()` method:

```suggestion
    if op.arg.dtype.is_timestamp():
```

**Review by cpcloud — CHANGES_REQUESTED** [review]

File: ibis/backends/flink/registry.py (hunk 1 above)
Comment (mfatihaktas) [reply ibis/backends/flink/registry.py:56]: Thanks, I replaced `isinstance()` with `is_*` in all my changes.

File: ibis/backends/flink/registry.py (hunk 2 above)
Comment (mfatihaktas) [reply ibis/backends/flink/registry.py:280]: Ah, sorry, I missed it again, even after reviewing my own code :) Thank you for catching it.

File: ibis/backends/flink/registry.py (hunk 3 above)
Comment (mfatihaktas) [reply ibis/backends/flink/registry.py:284]: Thanks for the reminder, fixed it.

File: ibis/backends/flink/registry.py (hunk 4 above)
Comment (mfatihaktas) [reply ibis/backends/flink/registry.py:325]: Thanks, fixed.

**Review by cpcloud — APPROVED** [review]

**Pull request merged**